### DISCUSSION TITLE
Units and Axis

### DISCUSSION THREAD

**Discussion by wallworm** [Autodesk/maya-usd#3184] [Q&A]
We are planning on adding better support for axis and units in MayaUSD. That includes any place that this needs to be considered:

- Export
- Import
- Stage in USD Proxy Shape

Among the things to solve will be giving pipelines and users to control how MayaUSD translates and uses units and axis data. One goal is to make it so that the user does not have to use specific settings to keep a stage oriented or scaled correctly as well as allow studios/pipelines to be able to control how these features work if there are specific requirements.

Some use cases we intend to solve:
- Loading any USD file created in Z-up environments (for example the Kitchen Set) do not come in on its side
- Loading scenes of various scene scales can remain consistent across different DCCs
- Controlling the exchange of units/axis in exporters/importers

We also intend to provide:
- Providing a UI for saving user preferences on all scale and axis options to set user defaults in new scenes
- Provide ENV variables that allow pipelines to enforce certain rules

We have not started this work but want to open it up for discussion for the community.

**Comment by dgovil** [comment]
This is great to hear.

For export:
I think exposing `metersPerUnit` as a UI option for Export would be a good start, maybe as a `Units: <dropdown-of-MUnit-enum>`. Granted it only supports a few object types, but it still feels like it would be handy. It should be fairly easy to start growing the support across the other dimensioned unit types.

For Import:
It would be great if things could be implicitly rotated and scaled to match the Maya scene, or at least have an option for it. I believe we already added support for frame rate conformance, so axis and units conformance would be really welcome.

**Reply by neilh-adsk** [reply]
When would you want to honour the directive specified but the file itself, or would you always want to match the current Maya scene.
